upload_log_files.py

```diff
@@ -0,0 +1,139 @@
+    pipeline_configuration_file_path = args.pipeline_configuration_file_path
+    memory_profile_dir_path = args.memory_profile_dir_path
+    data_archive_dir_path = args.data_archive_dir_path
+
+    date_pattern = r'\d{4}-\d{2}-\d{2}'
+
+    log.info("Loading Pipeline Configuration File...")
+    with open(pipeline_configuration_file_path) as f:
+        pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+    Logger.set_project_name(pipeline_configuration.pipeline_name)
+    log.debug(f"Pipeline name is {pipeline_configuration.pipeline_name}")
+
+    uploaded_memory_logs = google_cloud_utils.list_blobs(google_cloud_credentials_file_path,
+                                                         pipeline_configuration.memory_profile_upload_bucket,
+                                                         pipeline_configuration.bucket_dir_path, )
+    uploaded_memory_log_dates = get_uploaded_log_dates(uploaded_memory_logs, date_pattern)
+    uploaded_data_archives = google_cloud_utils.list_blobs(google_cloud_credentials_file_path,
+                                                           pipeline_configuration.memory_profile_upload_bucket,
+                                                           pipeline_configuration.bucket_dir_path)
+    uploaded_data_archives_dates = get_uploaded_log_dates(uploaded_data_archives, date_pattern)
+
+    log.warning(f"Deleting old memory profile files from local disk...")
+    delete_old_log_files(memory_profile_dir_path, uploaded_memory_log_dates)
+    log.warning(f"Deleting old data archives files from local disk...")
+    delete_old_log_files(data_archive_dir_path, uploaded_data_archives_dates)
+
+    for file in get_file_paths(memory_profile_dir_path):
+        file_date_match = re.search(date_pattern, file)
+        file_date = file_date_match.group()
+        if file_date in uploaded_memory_log_dates:
+            log.info(f"Memory profile file already uploaded for {file_date}, skipping...")
+        else:
+            latest_memory_log_file_path = fetch_latest_modified_file_path(memory_profile_dir_path)
+            memory_profile_upload_location = f"{pipeline_configuration.memory_profile_upload_bucket}/" \
+                f"{pipeline_configuration.bucket_dir_path}/{os.path.basename(latest_memory_log_file_path)}"
+            log.info(f"Uploading memory profile from {latest_memory_log_file_path} to {memory_profile_upload_location}...")
+            with open(latest_memory_log_file_path, "rb") as f:
+                google_cloud_utils.upload_file_to_blob(google_cloud_credentials_file_path, memory_profile_upload_location, f)
+
+
+    for file in get_file_paths(data_archive_dir_path):
+        file_date_match = re.search(date_pattern, file)
+        file_date = file_date_match.group()
+        if file_date in uploaded_data_archives:
+            log.info(f"Data archive file already uploaded for {file_date}, skipping...")
```

Comment:
Can you include the file_path you skipped here as a double-check when reading the logs

Reply:
updated thanks